plugin/cassandra/plugin.go

```diff
@@ -90,106 +90,151 @@ package main
 import (
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 
-	fmt "github.com/jhunt/go-ansi"
+	fmt "github.com/jhunt/go-ansi" 
 
 	"github.com/starkandwayne/shield/plugin"
 )
 
+// Default configuration values for the plugin
 const (
-	DefaultHost     = "127.0.0.1"
-	DefaultPort     = "9042"
-	DefaultUser     = "cassandra"
-	DefaultPassword = "cassandra"
-	DefaultDataDir  = "/var/vcap/store/cassandra/data"
-
+	DefaultHost      = "127.0.0.1"
+	DefaultPort      = "9042"
+	DefaultUser      = "cassandra"
+	DefaultPassword  = "cassandra"
+	DefaultSaveUsers = true
+	DefaultDataDir   = "/var/vcap/store/cassandra/deployment_name/data"
+	DefaultBinDir    = "/var/vcap/jobs/cassandra/bin"
 	VcapOwnership = "vcap:vcap"
 	SnapshotName  = "shield-backup"
 )
 
+// Array or slices aren't immutable by nature; you can't make them constant
+var (
+	DefaultExcludeKeyspaces = []string{"system_schema", "system_distributed", "system_auth", "system", "system_traces"}
+	SystemAuthTables        = []string{"roles", "role_permissions", "role_members", "resource_role_permissons_index"}
+)
+
 func main() {
 	p := CassandraPlugin{
 		Name:    "Cassandra Backup Plugin",
 		Author:  "Orange",
-		Version: "0.1.0",
+		Version: "1.0.0",
 		Features: plugin.PluginFeatures{
 			Target: "yes",
 			Store:  "no",
 		},
 		Example: `
 {
-  "host"         : "127.0.0.1",      # optional
-  "port"         : "9042",           # optional
-  "user"         : "username",
-  "password"     : "password",
-  "keyspace"     : "db",
-  "datadir"      : "/path/to/data"   # optional
+  "host"              : "127.0.0.1",      # optional
+  "port"              : "9042",           # optional
+  "user"              : "username",
+  "password"          : "password",
+  "include_keyspaces" : ["db"],
+  "exclude_keyspaces" : ["system"],
+  "save_users"        : true,
+  "bindir"            : "/path/to/bin",   # optional
+  "datadir"           : "/path/to/data",  # optional
 }
 `,
 		Defaults: `
 {
-  "host"     : "127.0.0.1",
-  "port"     : "9042",
-  "user"     : "cassandra",
-  "password" : "cassandra",
-  "datadir"  : "/var/vcap/store/cassandra/data"
+  "host"              : "cassandra_host", // "12.0.0.1"
+  "port"              : "9042", // "cassandra_port"
+  "user"              : "cassandra_user",     //cassandra
+  "password"          : "cassandra_password", // "cassandra"
+  "include_keyspaces" : [""],
+  "exclude_keyspaces" : [ "system_schema", "system_distributed", "system_auth", "system", "system_traces" ],
+  "save_users"        : true,
+  "bindir"            : "/var/vcap/jobs/cassandra/bin", 
+  "datadir"           : "/var/vcap/store/cassandra/data" 
 }
 `,
 		Fields: []plugin.Field{
-			plugin.Field{
+			{
 				Mode:     "target",
 				Name:     "host",
 				Type:     "string",
 				Title:    "Cassandra Host",
 				Help:     "The hostname or IP address of your Cassandra server.",
 				Default:  "127.0.0.1",
-				Required: true,
+				Required: false,
```

Comment:
If this is no longer required, we should remove the default.  If the default is standing in for "no value", then it should still be considered _required_.